cli/parse_test.ts

```diff
@@ -0,0 +1,725 @@
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("multiple same aliases", () => {
+    const expected = { options: { foo: true }, arguments: {} };
+    const actual = parse(["-f", "-f"], {
+      options: [
+        { name: "foo", alias: "f" },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("multiple combined aliases", () => {
+    const expected = { options: { verbose: 3 }, arguments: {} };
+    const actual = parse(["-vvv"], {
+      options: [
+        {
+          name: "verbose",
+          alias: "v",
+          default: 0,
+          fn: (value) => value += 1,
+        },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("multiple combined aliases with value", () => {
+    const expected = { options: { verbose: 3 }, arguments: {} };
+    const actual = parse(["-v", "3"], {
+      options: [
+        {
+          name: "verbose",
+          alias: "v",
+          type: Number,
+          default: 0,
+          value: { name: "VALUE" },
+        },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("name and alias", () => {
+    const expected = {
+      options: { foo: true, bar: true },
+      arguments: {},
+    };
+    const actual = parse(["--foo", "-b"], {
+      options: [
+        { name: "foo", alias: "f" },
+        { name: "bar", alias: "b" },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("name and same alias", () => {
+    const expected = { options: { foo: true }, arguments: {} };
+    const actual = parse(["--foo", "-f"], {
+      options: [
+        { name: "foo", alias: "f" },
+      ],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("default value", () => {
+    const expected = { options: { foo: "foo" }, arguments: {} };
+    const actual = parse(["--foo"], {
+      options: [{ name: "foo", default: "foo" }],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("mismatching type", () => {
+    assertThrows(() => {
+      parse(["--foo"], {
+        options: [{
+          name: "foo",
+          type: (Boolean as unknown as StringConstructor),
+          default: "foo",
+        }],
+      });
+    });
+  });
+  await t.step("name with value property", () => {
+    const expected = { options: { foo: "bar" }, arguments: {} };
+    const actual = parse(["--foo", "bar"], {
+      options: [{ name: "foo", value: { name: "VALUE" } }],
+    });
+    assertEquals(actual, expected);
+  });
+  await t.step("name without value property but present value", () => {
+    assertThrows(() => {
+      parse(["--foo=bar"], { options: [{ name: "foo" }] });
+    });
+  });
+  await t.step("name with multiple values", () => {
+    const expected = { options: { foo: ["bar", "baz"] }, arguments: {} };
+    const actual = parse(["--foo", "bar", "baz"], {
+      options: [{ name: "foo", value: { name: "VALUE", multiple: true } }],
+    });
```

Comment:
I would expect to require passing `--foo bar --foo baz` here. The current `parseArgs` works that way. 

Same with all the other cases with `multiple`.

Reply:
Fair point.